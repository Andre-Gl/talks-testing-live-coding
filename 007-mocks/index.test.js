jest.mock('./data')
const states = require('./index');

let statesAbbrList;

beforeEach(() => {
  statesAbbrList = states.getStatesAbbrList();
});

describe('getStatesAbbrList', () => {
  it('should return list of abbreviations', () => {
    expect(statesAbbrList).toMatchSnapshot();
  });
});

describe('getStatesTypeAhead', () => {
  it('should return empty array when no input', () => {
    const result = states.getStatesTypeAhead();
    expect(result).toMatchSnapshot();
  });
  it('should return array of states that starts with symbols', () => {
    const typeAheadInput = 'Test0';
    const result = states.getStatesTypeAhead(typeAheadInput);
    expect(result).toMatchSnapshot();
  });
  it('should return array of states that starts with symbols limitted to 3 matches', () => {
    const typeAheadInput = 'Test';
    const result = states.getStatesTypeAhead(typeAheadInput);
    expect(result).toMatchSnapshot();
  });
});

describe('zipLookup', () => {
  it('should resolve correct zip', () => {
    expect.assertions(1);
    return expect(states.zipLookup('94085')).resolves.toMatchSnapshot();
  });


  it('should reject on invalid zip', () => {
    expect.assertions(1);
    return expect(states.zipLookup('')).rejects.toMatchSnapshot();
  });
});
