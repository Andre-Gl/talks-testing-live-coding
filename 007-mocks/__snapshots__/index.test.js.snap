// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStatesAbbrList should return list of abbreviations 1`] = `
Array [
  "TST0",
  "TST1",
  "TST2",
  "TST3",
]
`;

exports[`getStatesTypeAhead should return array of states that starts with symbols 1`] = `
Array [
  "Test0",
]
`;

exports[`getStatesTypeAhead should return array of states that starts with symbols limitted to 3 matches 1`] = `
Array [
  "Test0",
  "Test1",
  "Test2",
]
`;

exports[`getStatesTypeAhead should return empty array when no input 1`] = `Array []`;

exports[`zipLookup should reject on invalid zip 1`] = `"Zip code is required"`;

exports[`zipLookup should resolve correct zip 1`] = `"TST"`;
